### PR TITLE
feat: add ignorePattern and ignore .d.ts by default

### PR DESCRIPTION
**Pull request by jvddavid**
I added these two features, as I was using the library, and when building my application it tried to load a .d.ts file, causing an error.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,38 +46,25 @@ jobs:
...
+                  node-version: 18
+                  registry-url: https://npm.pkg.github.com/
 
             # Show current package
             - name: cat package.json
               run: cat ./package.json
 
             # Install Dependencies
-            - run: npm ci
+            - run: npm i
```
Comment (GiovanniCardamone) [inline .github/workflows/ci.yml]: i think this should be `ci` as previously

**Review by GiovanniCardamone — COMMENTED** [review]
Thanks for your contributions, can you fix the installation step?

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+@jvddavid:registry=https://npm.pkg.github.com
```
Comment (GiovanniCardamone) [inline .npmrc]: can you remove this file?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,9 @@
 {
-    "name": "fastify-autoroutes",
-    "version": "3.0.2",
+    "name": "@jvddavid/fastify-autoroutes",
```
Comment (GiovanniCardamone) [inline package.json]: can you revert to previous package name?

**Comment by GiovanniCardamone** [comment]
also i would like to keep previous node version, there is no engine dependency in fastify package, but needs to be compatible from >= 14

**Comment by jvddavid** [comment]
I reverted to just the necessary changes, now you can change your git workflow however you want.

File: package.json (hunk 3 above)
Comment (jvddavid) [reply package.json]: yes, i changed it just to use at the moment

File: .npmrc (hunk 2 above)
Comment (jvddavid) [reply .npmrc]: yes

**Comment by GiovanniCardamone** [comment]
Hello, sorry for the late response, after thinking a bit about this, I don't think it's a necessary change.

The routes directory should only contain the routes modules (and maybe tests). You don't need to interact with the types of this because it's loaded through the module itself.

If you want to have types guaranteed in the route, then you need typescript.

So I'll reject this PR. But thanks for the time and effort you put into this

**Pull request closed without merging**